### PR TITLE
Updated syntax to modern standards

### PR DESCRIPTION
**Pull request by jwlawrence**
- Converted to classes and decorators
- Replaced component lifecycle hooks with ember-render-modifiers
- Use this. rather than get
- Introduced some quality of life babel plugins
- Implemented angle bracket components

### CODE REVIEW

**Comment by jwlawrence** [comment]
Before I look at the failures in older LTS versions, do we still want to support them?

File: addon/components/flash-message.js
Comment (luketheobscure) [inline addon/components/flash-message.js:12]: Is this just to make it easier to transition to glimmer components? Might be a breaking change for some consumers with little benefit.

File: addon/components/flash-message.js
Comment (luketheobscure) [inline addon/components/flash-message.js:33]: Missing a `messageStyle` dependency? Newer versions of eslint-plugin-ember can find and fix these kinds of errors.

File: addon/components/flash-message.js
Comment (luketheobscure) [inline addon/components/flash-message.js:99]: Is there an advantage to doing this vs. using the `@action` decorator?

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:12]: Yeah, you're probably right, I didn't run try before I opened the PR so I didn't realize how many older versions broke. I would attempt to move this work to the 2.0 branch, but that hasn't been touched in four years. I was attempting to breathe some life into this project, but seems like without the help of the maintainers it will be an uphill battle. I'll close this PR.

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:33]: Not sure what you mean? I don't see the issue?

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:99]: This function does use the `@action` decorator and is called on "init" via the `did-insert` helper in the template.

File: addon/components/flash-message.js
Comment (luketheobscure) [reply addon/components/flash-message.js:12]: Sorry, I think this PR is 99% of the way there! Nearly all these changes are net wins. The only one I was worried about was this tagName decorator.

File: addon/components/flash-message.js
Comment (luketheobscure) [reply addon/components/flash-message.js:33]: `alertType` won't update when `this.messageStyle` changes. Just needs line 33 updated to `@computed('flash.type', 'messageStyle')`

File: addon/components/flash-message.js
Comment (luketheobscure) [reply addon/components/flash-message.js:99]: Should be able to get rid of the `.bind` stuff then. `@action` does this for you!

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:33]: Oh gotcha, I didn't look at the body heh. I'll update computed and see if I can get the linter to pick up this issue.

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:12]: Ah, I suppose I could leave this PR open and leave it up to the maintainers to determine if they're ok breaking older versions, I just assumed that it would be a deal-breaker.

As for your concerns about making this tagless, I'll point out that I added a wrapping div to the template that implements all the bindings we had before, namely classnames and applying various attributes, which is now handled via `...attributes`.

That said, this would introduce a change in markup in apps that do provide a custom tag type to the component and could potentially break them. I think going forward this approach is more in line with where we're heading with glimmer components, but I can remove this change if you think it's going to break consumers.

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:99]: Good call 👍

File: addon/components/flash-message.js
Comment (luketheobscure) [reply addon/components/flash-message.js:12]: I don't have any data to back it up, but I have a hunch that customizing the tagName is pretty rare. Adding the div in the template satisfies my main concern which was breaking styles that relied on that HTML structure.

File: addon/components/flash-message.js
Comment (luketheobscure) [inline addon/components/flash-message.js:101]: I'm not positive, but I think you might need the `@action` decorator on `this._mouseEnter` and `this._mouseLeave` to preserve the context. Other than that this all looks great!

**Review by luketheobscure — APPROVED** [review]

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:12]: Yeah, that was my thought too. If a user does use a custom tag it would change their markup only slightly, but it should be an easy fix (famous last words).

File: addon/components/flash-message.js
Comment (jwlawrence) [reply addon/components/flash-message.js:101]: Since the `@action` decorator is used on `onDidInsert` the references in its body will be bound to the component context, and since the handler functions are internal and not mutable (shouldn't be anyway) I don't think we should have an issue leaving them as "private" functions without the action decorator. I could be mistaken though, nothing is ever straightforward when dealing with `this`. The dummy app seems to behave as expected though and all tests pass.

File: package.json
Comment (sbatson5) [inline package.json:3]: Can we not include a version bump here?  Since we drop support for older versions of ember and require newer versions of node, I would consider this a major version.  I can cut that as a separate PR

**Review by sbatson5 — COMMENTED** [review]
Thank you for taking the time to this 🙌 

Just one comment about the version number

**Pull request merged**
